### PR TITLE
docs: Theme Editor

### PR DESCRIPTION
**Pull request by MarcinVaadin**
Documentation for theme editor

### CODE REVIEW

File: articles/tools/theme-editor/getting-started.asciidoc [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+== Picking components
+
+To begin using Theme Editor pick component you want to style.
+
+image::images/pick-component-2.png[Pick an element to get started]
+
+Component picker allows navigating through different layers of layout allowing to pick exact component.
+
+=== Vaadin components
```
Comment (MarcinVaadin) [inline articles/tools/theme-editor/getting-started.asciidoc:28]:
```suggestion
=== Vaadin Components
```

File: articles/tools/theme-editor/getting-started.asciidoc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+=== Local
+
+image::images/local-classname.png[Setting local CSS class name]
+
+In Local mode, Theme Editor uses unique class name to distinguish picked component.
+If there is none existing yet, Theme Editor suggests new.
+
+[NOTE]
+Class name is applied to component using `addClassName` method call with `<theme-editor-local-classname>` annotation.
+Theme Editor will lose track of component if modified manually.
```
Comment (MarcinVaadin) [inline articles/tools/theme-editor/getting-started.asciidoc:65]:
```suggestion
Theme Editor lose track of component if modified manually.
```

File: articles/tools/theme-editor/index.asciidoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+Work with single component::
+Style single component automatically applying required code updates.
+
+Work with multiple components at once::
+Style all components of given type creating universal style rules.
+
+Easily find changes::
+Go to line where change has been applied with single click.
+
+<<getting-started#,Get started using Theme Editor>>
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply articles/tools/theme-editor/index.asciidoc:34]: Leaving for @jouni or @russelljtdyer  to comment

File: articles/tools/theme-editor/index.asciidoc [hunk 4]
```diff
@@ -0,0 +1,37 @@
+---
+title: Theme Editor
+section-nav: commercial
+---
+
+[[theme-editor.overview]]
+= Theme Editor (Free Preview)
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply articles/tools/theme-editor/index.asciidoc:7]: Added but not sure in proper place, @jouni please verify

**Review by Mikaelsu — APPROVED** [review]

**Pull request merged**